lib/tools/android-manifest.js

```diff
@@ -74,4 +81,60 @@ manifestMethods.packageAndLaunchActivityFromManifest = async function (localApk)
   }
 };
 
+manifestMethods.compileManifest = async function (manifest, manifestPackage, targetPackage) {
+  log.debug(`Compiling manifest ${manifest}`);
+  let {platform, platformPath} = await getAndroidPlatformAndPath();
+  if (!platform) {
+    return new Error("Required platform doesn't exist (API level >= 17)");
+  }
+  log.debug('Compiling manifest.');
+  try {
+    let args = ['package', '-M', manifest, '--rename-manifest-package',
+                manifestPackage, '--rename-instrumentation-target-package',
+                targetPackage, '-I', path.resolve(platformPath, 'android.jar'),
+                '-F', manifest + '.apk', '-f'];
+    await exec(this.binaries.aapt, args);
+    log.debug("Compiled manifest");
+  } catch (err) {
+    throw new Error(`Error compiling manifest. Original error ${err.message}`);
+  }
+};
+
+manifestMethods.insertManifest = async function (manifest, srcApk, dstApk) {
+  log.debug(`Inserting manifest, src: ${srcApk} dst: ${dstApk}`);
+  try {
+    await unzipFile(`${manifest}.apk`);
+    await ncp(srcApk, dstApk);
+    log.debug("Testing new tmp apk.");
+    await assertZipArchive(dstApk);
+    log.debug("Moving manifest.");
+    if (system.isWindows()) {
+      let java = path.resolve(process.env.JAVA_HOME, 'bin', 'java');
+      let args = ['-jar',  path.resolve(helperJarPath, 'move_manifest.jar'),
+                  dstApk, manifest];
+      await exec(java, args);
+    } else {
+      // Insert compiled manifest into /tmp/appPackage.clean.apk
+      // -j = keep only the file, not the dirs
+      // -m = move manifest into target apk.
+      await exec('zip', ['-j', '-m', dstApk, manifest]);
+    }
+    log.debug("Inserted manifest.");
+  } catch(e) {
+    throw new Error(`Error insterting manifest. Original error: ${e.message}`);
+  }
+};
+
+manifestMethods.hasInternetPermissionFromManifest = async function (localApk) {
+  try {
+    await this.initAapt();
+    log.debug("Checking if has internet permission from manifest.");
+    let {stdout} = await exec(this.binaries.aapt, ['dump', 'badging', localApk]);
+    return new RegExp("uses-permission:.*'android.permission.INTERNET'").test(stdout);
```

Comment:
I don't know about anyone else, but I'd prefer regexp literals where it's not dynamic. That way everything is at load-time, not run-time.